FloripaPublicTransportation/RestApi.swift

```diff
@@ -0,0 +1,126 @@
+    /// - Parameter completion: the block to be called on asynchronous task completion. It should be such that an object receives the `departures` array.
+    static func findDeparturesByRouteId(routeId: Int, completion: (departures: [Departure]) -> Void)
+    {
+        var departures = [Departure]()
+        parseRequest(NSURL(string: "https://api.appglu.com/v1/queries/findDeparturesByRouteId/run")!, params: ["routeId":"\(routeId)"], taskCompletion: { jsonRows in
+            for row in jsonRows {
+                departures.append(Departure(calendar: row["calendar"] as! String, time: row["time"] as! String))
+            }
+            departures.sortInPlace({ d1, d2 in
+                (d1.time! < d2.time!)
+            })
+            completion(departures: departures)
+        })
+    }
+    
+    /// Generic function to be called by any of the API POST methods.
+    /// The request is built upon the `url` and `params` arguments;
+    /// the `jsonRows` returned from the JSON response can be parsed via the `taskCompletion` function argument.
+    private static func parseRequest(url: NSURL, params: NSDictionary, taskCompletion: (jsonRows: [NSDictionary]) -> Void)
+    {
+        let request = NSMutableURLRequest(URL: url)
+        request.HTTPMethod = "POST"
+        request.addValue("staging", forHTTPHeaderField: "X-AppGlu-Environment")
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        
+        let username = "WKD4N7YMA1uiM8V"
+        let password = "DtdTtzMLQlA0hk2C1Yi5pLyVIlAQ68"
+        let credentialsData = "\(username):\(password)".dataUsingEncoding(NSUTF8StringEncoding)
+        let credentialsBase64String = credentialsData?.base64EncodedStringWithOptions([])
+        request.setValue("Basic \(credentialsBase64String!)", forHTTPHeaderField: "Authorization")
+        
+        var parameters: [String: AnyObject] = [:]
+        parameters["params"] = params
+        do {
+            request.HTTPBody = try NSJSONSerialization.dataWithJSONObject(parameters, options: [])
+        }
+        catch let error as NSError {
+            NSLog("NSJSONSerialization.dataWithJSONObject threw an error: \(error.description)")
+            NSLog("Dictionary contents: \(params)")
+        }
+        
+        let session = NSURLSession.sharedSession()
+        let task = session.dataTaskWithRequest(request,
+            completionHandler: { (data, response, error) in
+                guard error == nil else {
+                    NSLog("dataTaskWithRequest sent an error: \(error!.description)")
+                    // calls taskCompletion with an empty NSDictionary:
+                    taskCompletion(jsonRows: [NSDictionary]())
+                    delegate?.onDone()
+                    return
+                }
+                
+                do {
+                    let json = try NSJSONSerialization.JSONObjectWithData(data!, options: .MutableLeaves) as? NSDictionary
```

Comment:
This is just Objective-C legacy, in Swift this option is not necessary anymore (strings are mutable), as you've correctly pointed out.